storage/remote/write.go

```diff
@@ -74,6 +69,14 @@ func NewWriteStorage(logger log.Logger, reg prometheus.Registerer, walDir string
 		flushDeadline:     flushDeadline,
 		samplesIn:         newEWMARate(ewmaWeight, shardUpdateDuration),
 		walDir:            walDir,
+		highestTimestamp: &maxGauge{
+			Gauge: prometheus.NewGauge(prometheus.GaugeOpts{
```

Comment:
since you're not using promauto here you need to register this metric

Reply:
With the current setup, I guess that caused panics due to duplicate registering. Hence, did a careful placement of this.

Reply:
Oh yes. That got missed in a mistake. Thanks for pointing it out.